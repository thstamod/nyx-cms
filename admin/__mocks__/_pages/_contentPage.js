export const sidebar = {
  data: {
    documentTypes: [
      {
        _id: '5e986ce27a141e6a5ca5f2ee',
        name: 'depth_0',
        __typename: 'DocumentType',
        descendants: [
          {
            documentType: {
              _id: '5e986cfb7a141e6a5ca5f2ef',
              name: 'depth_11',
              __typename: 'DocumentType',
              descendants: [],
            },
          },
          {
            documentType: {
              _id: '5e986d007a141e6a5ca5f2f0',
              name: 'depth12_with_child',
              __typename: 'DocumentType',
              descendants: [
                {
                  documentType: {
                    _id: '5e986d097a141e6a5ca5f2f1',
                    name: 'depth_2',
                    __typename: 'DocumentType',
                    descendants: [],
                  },
                },
              ],
            },
          },
        ],
      },
      {
        _id: '5e98743e707dfc01d6ce173e',
        name: 'depth0_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
      {
        _id: '5e989eee707dfc01d6ce173f',
        name: 'depth0_1_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
    ],
  },
};
export const panel = {
  data: {
    documentType: {
      _id: '5e98743e707dfc01d6ce173e',
      name: 'depth0_no child',
      parentDocumentType: null,
      compilation: [
        {
          dataType: {
            type: 'Text',
            name: 'text',
          },
          value: {
            val: 'I am a Title!!',
          },
          title: 'title',
          options: null,
        },
        {
          dataType: {
            type: 'TextArea',
            name: 'TextArea',
          },
          value: {
            val:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
          },
          title: 'big text',
          options: null,
        },
      ],
      __typename: 'DocumentType',
    },
  },
};
