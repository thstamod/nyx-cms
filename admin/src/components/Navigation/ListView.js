/* eslint-disable react/prop-types */
import React from 'react';
// import { NavLink } from 'react-router-dom';
// import styled from 'styled-components';
import { Nav } from 'react-bootstrap';
// import SubMenu from './SubMenu';

// const Sidebar = styled('div')`
//   background: #fff000;
//   width: 200px;
//   color: #fff;
// `;

const ListView = ({ data }) => {
  const showDocumentTypes = () => {
    console.log(data.documentTypes);
    const list = data.documentTypes.map((el) => (
      <span key={el._id}>{el.name}</span>
    ));
    return list;
  };
  return (
    <Nav defaultActiveKey="/home" className="flex-column">
      {/* <Nav.Link href="/home">Active</Nav.Link> */}
      {showDocumentTypes()}
    </Nav>
  );
};

export default ListView;
