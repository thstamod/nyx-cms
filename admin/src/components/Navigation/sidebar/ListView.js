/* eslint-disable react/prop-types */
import React from 'react';
// import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import { Nav } from 'react-bootstrap';
// import SubMenu from './SubMenu';

// const Sidebar = styled('div')`
//   background: #fff000;
//   width: 200px;
//   color: #fff;
// `;

const Text = styled('div')`
  font-family: ${(props) => props.theme.fonts[1]};
  background: #fff000;
  width: 200px;
  color: ${(props) => props.theme.colors.lightBlue};
`;

const ListView = ({ data }) => {
  const showDocumentTypes = () => {
    // console.log(theme);
    const list = data.documentTypes.map((el) => (
      <Text key={el._id}>{el.name}</Text>
    ));
    return list;
  };
  return (
    <Nav defaultActiveKey="/home" className="flex-column">
      {/* <Nav.Link href="/home">Active</Nav.Link> */}
      {showDocumentTypes()}
    </Nav>
  );
};

export default ListView;
