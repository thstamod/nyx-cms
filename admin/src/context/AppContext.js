import React, { createContext, useContext } from 'react';
import { userReducer, initialState } from '../redux/reducers/userReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const AppContext = createContext();

export function useAppState() {
  return useContext(AppContext);
}

export function AppStateProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(userReducer, initialState, middlewares);

  return <AppContext.Provider value={store}>{children}</AppContext.Provider>;
}
