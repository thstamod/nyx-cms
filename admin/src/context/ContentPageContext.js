import React, { createContext, useContext } from 'react';
import {
  initialState,
  contentPageReducer,
} from '../state/reducers/contentPageReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const ContentPageContext = createContext(null);

export function useContentPageState() {
  return useContext(ContentPageContext);
}

export function ContentPageProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(
    contentPageReducer,
    initialState,
    middlewares
  );

  return (
    <ContentPageContext.Provider value={store}>
      {children}
    </ContentPageContext.Provider>
  );
}
