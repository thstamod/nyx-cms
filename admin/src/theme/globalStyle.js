import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html,body {
    width:100%;
  height: 100vh;
  display: flex;
  font-family: Karla, serif;
  }
#root {
  width:100%;
  height:100%;
}

  h1,h2,h3,h4,h5 {
    font-family: Rubik, serif;
  }
`;

export default GlobalStyle;
