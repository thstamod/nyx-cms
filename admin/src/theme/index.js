const theme = {
  colors: {
    powderWhite: '#F2F3F4',
    foggyOrange: '#D6804D',
    lightPurple: '#B886A5',
    darkOrange: '#AB5C49',
    darkGrey: '#262728',
  },
  fonts: ['Karla,sans-serif', 'Rubik,sans-serif'],
  fontSizes: {
    small: '1em',
    medium: '2em',
    large: '3em',
  },
};

export default theme;
