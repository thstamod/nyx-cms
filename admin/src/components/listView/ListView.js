import React, { useCallback } from 'react';
import ListViewItem from './ListViewItem';
import { useContentPageState } from '../../context/ContentPageContext';
import { selectDocType } from '../../state/actions/contentPageActions';

const ListView = ({ data }) => {
  const [state, dispatch] = useContentPageState();

  const handleListViewItemClick = useCallback(
    (id) => state.id !== id && dispatch(selectDocType(id)),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [state.id]
  );
  const showListViewItems = () => {
    if (!data) {
      return null;
    }

    const list = data.documentTypes.map(
      (el) =>
        el && (
          <ListViewItem
            data-testid="content-list-menu-item"
            key={el._id}
            data={el}
            handleClick={handleListViewItemClick}
          />
        )
    );
    return list;
  };
  return <div data-testid="content-list-menu">{showListViewItems()}</div>;
};

export default ListView;
