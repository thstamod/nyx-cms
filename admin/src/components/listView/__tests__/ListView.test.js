import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../../../theme';
import { ContentPageProvider } from '../../../context/ContentPageContext';
import ListView from '../ListView';
import { sidebar } from '../../../../__mocks__/_pages/_contentPage';

afterEach(cleanup);

test('ListView renders', () => {
  const { asFragment } = render(
    <ContentPageProvider>
      <ThemeProvider theme={theme}>
        <ListView data={sidebar.data} />
      </ThemeProvider>
    </ContentPageProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('with no data', () => {
  const { getByTestId } = render(
    <ContentPageProvider>
      <ThemeProvider theme={theme}>
        <ListView data={null} />
      </ThemeProvider>
    </ContentPageProvider>
  );

  expect(getByTestId('content-list-menu')).toBeEmpty();
});
