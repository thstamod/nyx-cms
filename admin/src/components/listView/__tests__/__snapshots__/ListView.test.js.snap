// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-column nav"
  >
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Home
      </span>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        About
      </span>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        About2
      </span>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        About3
      </span>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div />
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div />
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div />
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div>
        <div
          class="sc-AxjAm huDVTX"
        >
          Home
        </div>
        <div
          class="sc-AxjAm huDVTX"
        >
          About
        </div>
      </div>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div>
        <div
          class="sc-AxjAm huDVTX"
        >
          Home
        </div>
        <div
          class="sc-AxjAm huDVTX"
        >
          About
        </div>
      </div>
    </div>
    <div
      class="sc-AxirZ eOuGKq"
    >
      <span>
        Test
      </span>
      <div>
        <div
          class="sc-AxjAm huDVTX"
        >
          Home
        </div>
        <div
          class="sc-AxjAm huDVTX"
        >
          About
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
